keycloak/keycloak-operator
# KEYCLOAK-19101 Added new version folder 15.0.2 for the OLM deploy

Added label for the realm example to allow all-default CR creation

Signed-off-by: Jonathan Vila <jvilalop@redhat.com>

## JIRA ID
KEYCLOAK-19101

## Verification Steps

1. create a test catalog bundle to install this test operator ( from the deploy/olm-catalog/keycloak-operator/15.0.2 folder )
```bash
docker build -f bundle.Dockerfile -t keycloak-operator-bundle:1502 .

docker tag keycloak-operator-bundle:1502 quay.io/{your account}/keycloak-operator-bundle:1502

docker push quay.io/{your account}/keycloak-operator-bundle:1502

~/{your path to operator-registry project}/operator-registry/bin/opm index add --bundles quay.io/{your account}/keycloak-operator-bundle:1502 \
--tag quay.io/{your account}/keycloak-operator-test-catalog:1502 --container-tool docker --from-index quay.io/openshift-community-operators/catalog:latest

docker push quay.io/{your account}/keycloak-operator-test-catalog:1502

```
2. create an Openshift 4.7 instance using the Katacoda [playground](https://www.katacoda.com/courses/openshift/playgrounds/openshift47)
3. on the katacoda command line tab : vi catalog-source.yaml
```bash
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: test-catalog
  namespace: openshift-marketplace
  labels:
    application: keycloak
spec:
  sourceType: grpc
  image: quay.io/{your account}/keycloak-operator-test-catalog:{version}
   ```
4. on the katacoda command line tab 
```bash
oc login -u admin -p admin
sed "s/{version}/1502/g" catalog-source.yaml | kubectl apply -f -
```
5. install the operator from the OperatorHub screen in Openshift using the one on the test-catalog
6. create the keycloak, realm and client CRs with the default information
7. log to the keycloak admin ( using the route in the namespace and the secret to have the credentials ) and check that for the realm created there's the client created

## Checklist:
